Servers/UI/OJS.Servers.Ui/ClientApp/src/components/modal/ContestModal.tsx

```diff
@@ -0,0 +1,109 @@
+    const navigate = useNavigate();
+    const { actions: { setIsUserParticipant } } = useCurrentContest();
+
+    const startContestAndHideModal = useCallback(
+        () => {
+            setIsUserParticipant(true);
+
+            navigate(getParticipateInContestUrl({
+                id: contest.id,
+                participationType: ContestParticipationType.Compete,
+                problemIndex: 1,
+            }));
+
+            toggle();
+        },
+        [ contest, getParticipateInContestUrl, navigate, toggle, setIsUserParticipant ],
+    );
+
+    return isShowing
+        ? (
+            <div>
+                <Modal
+                  open={isShowing}
+                  sx={{ '& .MuiBackdrop-root': { backgroundColor: 'transparent' }, backdropFilter: 'blur(5px)' }}
+                  onClose={() => toggle()}
+                  aria-labelledby="modal-modal-title"
+                  aria-describedby="modal-modal-description"
+                >
+                    <Box className={styles.modal}>
+                        <Typography id="modal-modal-title" variant="h6" className={styles.modalHeading}>
+                            <>
+                                Starting now you will have
+                                {' '}
+                                {contest.duration}
+                                {' '}
+                                hours to complete the contest
+                                {' '}
+                                <span className={styles.bolded}>{contest.name}</span>
+                                .
+                            </>
+                        </Typography>
+                        <Typography id="modal-modal-description">
+                            Your time will start counting down when you press the &quot;Compete&quot; button.
+                            <br />
+                            <br />
+                            In the case of unexpected problems (turning off your computer, exiting the page/system,
+                            internet connection failure),
+                            {' '}
+                            <span className={styles.bolded}>the time lost will not be restored</span>
+                            . When time runs out,
+                            you
+                            {' '}
+                            <span className={styles.bolded}>will not</span>
+                            {' '}
+                            be able to compete in this competition again.
+                            <br />
+                            <br />
+                            When you click the &quot;Compete&quot; button,
+                            {' '}
+                            <span className={styles.bolded}>
+                                {contest.numberOfProblems}
+                                {' '}
+                                random problems, one of each type, will be generated
```

Comment:
If the modal is shown to all contests, this means we should have different message for Non-online exams, where random problems are NOT assigned. Or we can just show the modal for online contests..

Reply:
I changed it so that the modal appears only if contest is official and online.